python/ray/autoscaler/v2/instance_manager/common.py

```diff
@@ -223,7 +224,15 @@ def get_valid_transitions() -> Dict[
             Instance.TERMINATING: {
                 # When a cloud instance no longer appears in the list of running cloud
                 # instances from the node provider.
-                Instance.TERMINATED
+                Instance.TERMINATED,
+                # When the cloud instance failed to be terminated.
+                Instance.TERMINATION_FAILED,
```

Comment:
Added this so that we could seperate:
1. discovering termination failure 
2. retry termination.